MdeModulePkg/Include/Protocol/UsbAssociationIo.h

```diff
@@ -0,0 +1,81 @@
+/** @file
+  EFI Usb Interface Association protocol.
+
+  This protocol is used by USB drivers, running in the EFI boot services
+  environment to access USB devices that implement their configurations
+  using interface association: USB cameras, USB audio devices, USB modems, etc.
+
+  Copyright (c) 2025, American Megatrends International LLC. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#ifndef __USB_ASSOCIATION_IO_H__
+#define __USB_ASSOCIATION_IO_H__
+
+#include <IndustryStandard/Usb.h>
+#include <Protocol/UsbIo.h>
+
+typedef USB_INTERFACE_ASSOCIATION_DESCRIPTOR EFI_USB_INTERFACE_ASSOCIATION_DESCRIPTOR;
+
+//
+// Global ID for the USB IAD Protocol
+//
+#define EDKII_USB_IA_PROTOCOL_GUID \
+  { \
+    0xf4279fb1, 0xef1e, 0x4346, { 0xab, 0x32, 0x3f, 0xe3, 0x86, 0xee, 0xb4, 0x52 } \
+  }
+
+typedef struct _EDKII_USB_IA_PROTOCOL EDKII_USB_IA_PROTOCOL;
+
+/**
+  Get the USB Interface Association Descriptor from the current USB configuration.
+
+  @param[in]  This              A pointer to the EFI_USB_IA_PROTOCOL instance.
+  @param[out]  Descriptor       A pointer to the caller allocated USB Interface Association Descriptor.
+
+  @retval EFI_SUCCESS           The descriptor was retrieved successfully.
+  @retval EFI_INVALID_PARAMETER Descriptor is NULL.
+
+**/
+typedef

```

Comment:
Can these be library services instead of a new protocol?  If it is a new protocol, then USB Device Drivers will depends on EDK II specific protocols instead of just UEFI Protocols.

Reply:
In this PR we create USB Association handle and install DP and Association protocols on it. This handle is essential If USB Device Driver follows UEFI driver model, it can not be implemented using the library.
If this EDKII protocol is successful, it can be adopted by UEFI specification.
USB Association is a part of USB specification and this PR is an attempt to implement its functionality.
If there is any other way of advertising the new protocol, please advise.

Reply:
If the intent is to add to UEFI Spec, then there is a process for that and the doc and code updates would be done in edk2-staging repo.

https://github.com/tianocore/tianocore.github.io/wiki/EDK-II-Code-First-Process

Reply:
Thank you, will move it to edk2-staging. The document above talks about creating a bugzilla issue, I understand bugzilla function has been replaced with github issue. Should the issue be created in edk2 repo or edk2-staging repo?

Reply:
good question.  Please create in edk2 repo because that is the location the feature will be added once it is available in published spec.